File: proxy/api/_lib/contract.ts
/**
 * Shared API contract types and constants for the translator proxy.
 * Single source of truth for the `POST /api/translate` contract.
 */

export const MAX_TEXT_LENGTH = 500;

export const SUPPORTED_LANGUAGES = [
  "Spanish",
  "French",
  "Japanese",
  "German",
] as const;

export type SupportedLanguage = (typeof SUPPORTED_LANGUAGES)[number];

export interface TranslateRequest {
  text: string;
  targetLanguage: string;
}

export interface TranslateResponse {
  translation: string;
  pronunciation: string;
  context: string;
}

export interface ErrorResponse {
  error: string;
}

export type HandlerResult =
  | { ok: true; status: 200; body: TranslateResponse }
  | { ok: false; status: number; body: ErrorResponse };

File: android/app/src/main/java/com/example/voicetexttranslator/data/Contract.kt
package com.example.voicetexttranslator.data

/**
 * Shared API contract types for `POST /api/translate`, mirroring the proxy's
 * request/response shapes:
 *   - Request body:    { text, targetLanguage }
 *   - Success (200):   { translation, pronunciation, context }
 *   - Error (non-2xx): { error }
 */

data class TranslateRequest(
    val text: String,
    val targetLanguage: String,
)

data class TranslateResponse(
    val translation: String,
    val pronunciation: String,
    val context: String,
)

data class ApiError(
    val error: String,
)

/** Languages the translator supports. [label] is sent to the proxy as `targetLanguage`. */
enum class SupportedLanguage(val label: String) {
    SPANISH("Spanish"),
    FRENCH("French"),
    JAPANESE("Japanese"),
    GERMAN("German"),
}

/** Maximum length of translatable input text. */
const val MAX_TEXT_LENGTH: Int = 500
